### PR TITLE
Convert e2eTests to Java

### LINKED ISSUE #1057: Investigate Using Playwright's Java Interface for E2E Tests
[Playwright has a Java interface](https://playwright.dev/java/docs/intro). Given that the rest of the repo's tests and code is in Java (excluding the EDSL/Sequencing Server), it is worth investigating if it would be easier to write our Playwright tests in Java as well.

### PR DESCRIPTION
**Pull request by Mythicaeda**
* **Tickets addressed:** Closes #1057 
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description

Advantages to moving the e2eTests to Java:
- We now have the option to combine the DBTests with the rest of the e2eTests.
- We are now able to parallelize the e2eTests (Currently per-file parallelization is definitely supported, according to the [docs](https://playwright.dev/java/docs/test-runners#running-tests-in-parallel). Parallel Nested classes aren’t currently supported, since most of the nested classes are sharing a Playwright instance, but this could be altered. Our e2eTests are the slowest tests, so this is especially something worth looking into
- Easier to run/debug tests
- Minor, but we don't have to explicitly typecheck responses (`expect().notNull;`, `expect().toBe(number);` etc.)
- Minor, but developers don’t have to be familiar with TS to write/update e2eTests
   - Related, I think we’re generally more familiar with JUnit vs the Playwright TS interface

Outside of the conversion, the restructure from serial tests with side effects to isolated tests with proper Before and After tasks made it very easy to add new tests.

Disadvantages:
- Everything was a JSON in TS, which meant we could just define types and then declare the returns from Hasura requests as those types. In Java, we need to explicitly convert to and from JSON.

## Verification

Test updates beyond the conversion are:
- Tests hitting the deprecated `GetEffectiveActivityArguments` endpoint have been updated to hit the bulk endpoint
- Scheduling e2eTests have been fleshed out
- Model Effective Arguments tests have been added

## Documentation

The Readme in the e2eTest directory has been updated.

## Future work

I'd like to move the DBTests to be with the rest of the E2ETests, or at least have the them share some types.

### CODE REVIEW

File: e2e-tests/README.md [hunk 1]
```diff
@@ -15,5 +15,5 @@ docker compose -f docker-compose-test.yml up --build
 After starting the development Docker compose stack the following commands can be issued to run these tests:
 
 ```sh
-npm install && npm test
+./gradlew e2e-test:e2eTest
```
Comment (Mythicaeda) [inline e2e-tests/README.md]: Typo: should be `e2e-tests`

File: e2e-tests/src/test/java/gov/nasa/jpl/aerie/e2e/types/ExternalDataset.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+   * Input definition of a Profile
+   * @param name Name of the resource profile
+   * @param type Type of the profile ("real" or "discrete")
+   * @param schema Value Schema of the profile
+   * @param segments Profile Segments that constitute this profile
+   */
+  public record ProfileInput(
+      String name,
+      String type,
+      JsonObject schema,
```
Comment (Mythicaeda) [inline e2e-tests/src/test/java/gov/nasa/jpl/aerie/e2e/types/ExternalDataset.java]: Make toJson for ValueSchema

File: e2e-tests/src/test/java/gov/nasa/jpl/aerie/e2e/types/Plan.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+          simulation_end_time
+          template: simulation_template {
+            arguments
+            description
+            id
+          }
+        }
+      }
+    }
+   */
```
Comment (Mythicaeda) [inline e2e-tests/src/test/java/gov/nasa/jpl/aerie/e2e/types/Plan.java]: Remove

File: e2e-tests/src/test/java/gov/nasa/jpl/aerie/e2e/utils/HasuraRequests.java [hunk 4]
```diff
@@ -0,0 +1,664 @@
...
+  }
+
+  public int createPlan(
+      int modelId,
+      String name,
+      String duration,
+      String startTime,
+      Map<String, String> headers)
+  throws IOException {
+    final var modelInsertInput = Json.createObjectBuilder()
```
Comment (Mythicaeda) [inline e2e-tests/src/test/java/gov/nasa/jpl/aerie/e2e/utils/HasuraRequests.java]: Typo: planInsertInput

File: e2e-tests/src/test/java/gov/nasa/jpl/aerie/e2e/utils/HasuraRequests.java [hunk 5]
```diff
@@ -0,0 +1,664 @@
...
+            .add("model_id", modelId)
+            .add("name", name)
+            .add("duration", duration)
+            .add("start_time", startTime);
+    final var variables = Json.createObjectBuilder().add("plan", modelInsertInput).build();
+    return makeRequest(GQL.CREATE_PLAN, variables, headers).getJsonObject("insert_plan_one").getInt("id");
+  }
+
+  public Plan getPlan(int planId) throws IOException {
+    // TODO: Refine query based on usage
```
Comment (Mythicaeda) [inline e2e-tests/src/test/java/gov/nasa/jpl/aerie/e2e/utils/HasuraRequests.java]: Remove TODO

**Review by adrienmaillard — APPROVED** [review]
Great work !

**Review by mattdailis — APPROVED** [review]

**Pull request merged**